*** ip_var.h.orig	Sat Mar 22 15:06:28 2014
--- ip_var.h	Sat Mar 22 16:07:36 2014
***************
*** 158,163 ****
--- 158,170 ----
  #define IP_ROUTETOIF		SO_DONTROUTE	/* 0x10 bypass routing tables */
  #define IP_ALLOWBROADCAST	SO_BROADCAST	/* 0x20 can send broadcast packets */
  
+ #ifdef  LISP
+ #define IP_LISP                 0x40            /* GgX - LISP encapsulated 
+                                                   * Do not check possible 
+                                                   * encapsulation again.
+                                                   */
+ #endif  /* LISP */
+  
  /*
   * mbuf flag used by ip_fastfwd
   */
